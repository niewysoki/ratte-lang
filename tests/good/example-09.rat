fn TestShadowingVariable() -> Void {
  let mut a: Int = 0;
  let x: Int = 10;

  while (a < 20) {
    let x: Int = 1;
    a = a + x;
  }

  Println(ShowInt(a));
  return;
}

fn TestShadowingFunction() -> Void {
  fn TestShadowingVariable() -> Void {
    Println("surprise");
    return;
  }

  TestShadowingVariable();
  return;
}

fn TestShadowingFunctionSelf() -> Void {
  fn TestShadowingFunctionSelf() -> Void {
    Println("no recursion");
    return;
  }

  Println("recursion?");
  TestShadowingFunctionSelf();
  return;
}

fn Main() -> Void {
  TestShadowingVariable();
  TestShadowingFunction();
  TestShadowingFunctionSelf();
  return;
}