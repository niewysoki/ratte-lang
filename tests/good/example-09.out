20
surprise
recursion?
no recursion
